mlflow/pyfunc/__init__.py

```diff
@@ -1688,8 +1860,23 @@ def spark_udf(
     converted to string. If the result type is not an array type, the left most column with
     matching type is returned.
 
-    NOTE: Inputs of type ``pyspark.sql.types.DateType`` are not supported on earlier versions of
-    Spark (2.4 and below).
+    .. note::
+        Inputs of type ``pyspark.sql.types.DateType`` are not supported on earlier versions of
+        Spark (2.4 and below).
+
+    .. note::
+        When using Databricks Connect to connect to a remote Databricks cluster,
+        the Databricks cluster must use runtime version >= 16, and when 'spark_udf'
+        param 'env_manager' is set as 'virtualenv', the 'prebuilt_env_path' param is
+        required to be specified.
+
+    .. note::
+        Please be aware that when operating in Databricks Serverless,
+        spark tasks run within the confines of the Databricks Serverless UDF sandbox.
+        This environment has a total capacity limit of 1GB, combining both available
+        memory and local disk capacity. Furthermore, there are no GPU devices available
+        in this setup. Therefore, any deep-learning models that contain large weights
+        or require a GPU are not suitable for deployment on Databricks Serverless.
```

Comment:
1GB disk space seems pretty low. What if I'm using a tensorflow model? TF requires > 1GB to install.

Reply:
Yes. then we have no way to support it in Serverless. We have to consider Serverless Ray (in our roadmap)

Reply:
and the 1 GB is total capacity of memory + local-disk , because the sandbox local disk is a virtual disk mapped to memory similar to /dev/shm

Reply:
and note the current implementation will save model in NFS (Note this doesn't occupy UDF sandbox local disk space), so only the model in-memory size matters (in-memory model + other processes memory usage can't exceed 1GB).